### DISCUSSION TITLE
2 preisgünstige Zeitfenster an einem Tag

### DISCUSSION THREAD

**Discussion by dancekid** [torbennehmer/hacs-e3dc#181] [Help]
Hallo liebe Community,
Ich habe ein S10 E mit 19,5 kWh Speicher und 3k max Ladung.
In HomeAssist habe ich testweise ein 6H Ladefenster eingestellt.
Zwei 3H Ladefenster wären preisgünstiger. 
3-4 Zeitfenster mit 2H sind auch gut.
Wie kann man das konfigurieren. Er soll ja nicht zweimal das gleiche Fenster vorschlagen.

Gruß Michael

**Comment by Michel1973** [comment]
Hallo Michael, wie ermittelst du derzeit das Zeitfenster mit den günstigsten Tarif?

Ich nutze die Epex Spot Integration. Hierzu gibt es auch noch den Epex Spot Switch. Dieser lässt sich so konfigurieren, dass du innerhalb eines bestimmten Zeitfenster die günstigsten X Stunden ermitteln lassen kannst. Der Switch ist ein Binary Sensor. Du hast die Möglichkeit, beliebig viele Sensoren zu erstellen.

Viele Grüße 
Michel

**Reply by dancekid** [reply]
Hallo Namensvetter,
ich nutze auch den Epex Spot.
Habe mir hier mehr E3DC Erfahrung erhofft.

Bisher habe ich E3DC Control als Container (aus dem Photovoltaik Forum) genutzt.

Der Sohn eines guten Freundes hat mir eine Version für Node Red erstellt.

Alle Varianten laufen auf ein Zeitfenster hinaus. E3DC Control berücksichtigt den Durchschnittsverbrauch und errechnet den Bedarf für teure Stunden.

Awattar hat am 25.09.24 z.B. 3 ganz teure und 3 ganz preiswerte Stunden.

Die Automation müsste also

1) ausrechnen bei wieviel Stunden die Differenz zwischen teuer und preiswert > 10 Cent ist

2) Die in (1) errechnete Stundenzahl als Zeitfenster aus den vorhandenen 24h ermitteln und dann kann man den durchschnittlichen Verbrauch dieses Fensters laden.

Nur wie setzt man das um ?

Habe jetzt mal beim Epex Spot nachgefragt. Vielleicht können wir was adaptieren.

Gruß Michael aus Falkensee

**Reply by Michel1973** [reply]
Guten Morgen,
Ich habe aus Neugierde einen Epex Spot Sensor mit deinen Angaben erstellt (siehe Bild)![image](https://github.com/user-attachments/assets/9c7d6806-79ad-4ae7-a429-917bc836229b)

Dieser hat zwei Zeitfenster für den 25.9.2024 ermittelt. Das erste Zeitfenster geht von 0:00 Uhr bis 4:00 Uhr. Das zweite von 11:00 Uhr bis 14:00 Uhr.

Vielleicht hilft dir das weiter. 

Viele Grüße 
Michel

**Reply by dancekid** [reply]
Hallo Michel (doch nicht Namensvetter, da ein a fehlt und ich zu müde war :-(),

Die E3DC Hauskraftwerke und meine Wärmepumpe brauchen immer etwas Auszeit nach der Arbeit.
Die Variante Intermittent hat hier durch Zufall funktioniert und zwei entfernte Zeitfenster genommen.
Das System könnte im schlimmsten Fall aber auch jede 2. Stunde preisgünstig finden.

Somit wäre die Vorgabe der gewünschten Zeitfenster mit je 2H oder so besser. 

Gruß Michael

**Comment by torbennehmer** [comment]
Generell ist das ja nur indirekt ein Thema der e3dc integration, Ich denke Michels Ansatz ist da nicht falsch. Ich habe bei mir die Tibber Integration laufen, lade aber nur selten aus dem Netz, wegen ein paar Cent Preisunterschied ist das ja noch keine gute Idee.

Wo ihr generell aufpassen müsst ist, dass die e3dc API die Anzahl der aktiven` Ladevorgang erlaubt (manual charging) beschränkt. Offiziell einmal täglich, ich bin mir da aber nicht ganz sicher, ob es nicht ein wenig öfter geht. Darauf verlassen könnt ihr euch aber nicht. Idealerweise also nur einmal machen.

Genügend PV Leistung vorausgesetzt ist es auch ein Ansatz, zu bestimmten Zeiten (z.B. wenn günstig) eine Entladesperre zu setzen, damit der Akku nicht zur Überbrückung herangezogen wird, wenn es günstig ist. Dann läuft der per PV gemütlich voll.

**Reply by neuromancer3142** [reply]
Kurze Info zum Thema manuelles Laden: Bei mir kann ich den Vorgang alle 2h und 15 min neu starten. Wenn ich den Akku aus dem Netz lade setze ich eine Energiemenge die geladen werden soll und eine maximale Leistung. Also z.B. 12.000Wh mit maximal 6000W Ladeleistung. Beim Start lasse ich einen Timer mit 2h15mins mitlaufen. Nach dessen Ablauf kann das nächste Ladefenster starten. Funktioniert bei mir seit längerem ohne Probleme.

**Reply by torbennehmer** [reply]
Danke für die Info, ich teste das mal, eventuell hat sich das verändert oder es ist Gerätespezifisch. Ich hatte das vor längerer Zeit mal getestet, da war die Zeitspanne definitiv höher. Wäre ja super, 2 1/4 h wäre glatt nutzbar.

**Comment by dancekid** [comment]
Hallo liebe Community,

Habe den Epex Sensor eben installiert.

Darin gibt es eine Möglichkeit einer Duration Entity.
Da könnte man die ermittelte Stundenzahl eingeben.

Habe direkt bei epex spot nachgefragt und melde mich wenn es Neuigkeiten gibt.

Michael

**Comment by Cheppetto** [comment]
There are addons for that ;)

you might have a look at this one:
[https://github.com/TheFes/cheapest-energy-hours](url) in general packing to much intelligence which is not integration related into an integration is normally not a good idea.

neither in the device integration nor in Epex Spot

Maintenance external dependencies ....
